### PR TITLE
Integrate ICCC with KiCad using Rust

### PR DESCRIPTION
**Pull request by twelho**
Initialize the next generation of ICCC (#4) now integrated directly into KiCad and written in Rust. You may define expressions for component attributes in the "Symbol Properties" view in KiCad, and ICCC Rust edition (nicknamed `evaluator`) will parse the EESchema (`.sch`) file of KiCad and update the values.

This work also unites the `parser` (#6) and  `evaluator` to facilitate code sharing, and updates the `.gitignore` to work better with multi-binary Cargo workspaces.

### CODE REVIEW

File: tools/kicad_rs/src/bin/parser.rs [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 use std::error::Error;
 use std::path::Path;
 use std::{env, fmt};
 
+use kicad_parse_gen::schematic as kicad_schematic;
+
+use kicad_rs::types::*;
+
 // Main function, can return different kinds of errors
 fn main() -> Result<(), Box<dyn Error>> {
```
Comment (luxas) [inline tools/kicad_rs/src/bin/parser.rs:11]: I'll move the logic from here to the library after this has been merged.

File: tools/kicad_rs/src/bin/evaluator.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+use evalexpr::Context;
+
+use kicad_rs::resolver;
+use kicad_rs::resolver::*;
+
+// use crate::resolver;
+
+// Main function, can return different kinds of errors
+fn main() -> Result<(), Box<dyn Error>> {
```
Comment (luxas) [inline tools/kicad_rs/src/bin/evaluator.rs:12]: Will figure out this code after merge, to use the parsing library.

File: .gitignore [hunk 3]
```diff
@@ -29,15 +37,13 @@ Cargo.lock
 
 # Temporary files
 *.000
-*.bak
```
Comment (luxas) [inline .gitignore:32]: Why remove these three?

**Review by luxas — APPROVED** [review]
Thanks! I'm happy to merge this now, and then we can continue to iterate.

File: .gitignore (hunk 3 above)
Comment (twelho) [reply .gitignore:32]: IDE warned about duplicate entries so I removed the less relevant duplicates.

**Pull request merged**